Iterative graph walk rework

chensasha opened the pull request.

chensasha committed:
Update debruijnnode.cpp

chensasha committed:
Update debruijnnode.cpp

asl commented on graph/debruijnnode.cpp:
```diff
@@ -417,28 +417,31 @@ void DeBruijnNode::labelNeighbouringNodesAsDrawn(int nodeDistance, DeBruijnNode
+    std::unordered_set<DeBruijnNode *> s2;
+    s1.insert(callingNode);
+    for (int depth = 0; depth <= nodeDistance; depth++) {
+        for (auto otherNode : s1) {
```
```suggestion
        for (auto *otherNode : s1) {
```

asl commented on graph/debruijnnode.cpp:
```diff
@@ -417,28 +417,31 @@ void DeBruijnNode::labelNeighbouringNodesAsDrawn(int nodeDistance, DeBruijnNode
+                else
+                    otherNode->getReverseComplement()->m_drawn = true;
+            }
+            for (auto m_edge : otherNode->m_edges) {
```
```suggestion
            for (auto *m_edge : otherNode->m_edges) {
```

chensasha committed:
Update on debruijnnode.cpp

В функции DeBruijnNode::labelNeighbouringNodesAsDrawn изменено условие проверки посещения данной вершины и добавлено условие проверки пустоты set, когда потомков нет.
